kpops/components/base_components/kubernetes_app.py

```diff
@@ -88,7 +87,6 @@ class Config(CamelCaseConfig, DescConfig):
 
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
-        self.__check_compatible_name()
```

Comment:
The problem is that the substitution happens after at least one init of the component and a placeholder of the form `${foo}` is not a valid Kubernetes name

Reply:
To solve it, I would either have to create a new parameter in the KubernetesApp or figure out a way to deduce from the existing arguments whether it's the final initialization of the component or not

Reply:
I don't have a precise answer for you. About the _final initialization_ problem I can only add that this could be a call  in `enrich_component` after the component is initialized

Reply:
the simplest solution I can think of is to add a generic `validate` method to the base component and call it from `enrich_component`. Of course it would be nicer if this could use Pydantic's builtin validation somehow

Reply:
I added a validator, but I honestly don't like the solution much. Unfortunately I cannot think of a better way and simpler way to do it.

One side effect is that now the generated values of the connector contain a useless field `validate_name`. I could combat this by moving the validate_name class attribute from KubernetesApp to PipelineComponent, but then it introduces a KubernetesApp-specific field to all components

Reply:
Currently, I made it so that the name is validated by default and you have to deliberately opt-out. The reason is simple - to avoid introducing bugs in the future by forgetting to request validation.

Reply:
do we have to have the `validate_name` attribute for the `root_validator` to work? the validation doesn't have to be opt-out, it should always be enabled.

Reply:
The problem is that when the name contains a placeholder, it is not valid. Explicitly adding placeholders to the allowed string format does not work, because it is not necessary that the placeholder would be substituted

Reply:
ok I see :/ so your proposed workaround would be to set `validate_name=False` if the user wants to use placeholders in the name of a kubernetes-app component?

Reply:
> if it's not possible to do it conveniently with Pydantic's builtin validators this could just be a custom method `validate` that checks it

The difference would only be that instead of being decorated with `@root_validator`, it would have to be called in `__init__` on itself. I suppose that would be a bit more efficient?

Reply:
isn't it easier to expand the `check_compatible_name` and then filter the place holders and proof validation ?

Reply:
@irux I am not sure if I understand, what do you mean with expand?

Reply:
What I mean is just add the edge case where it needs to handle place holders and it filters them.

Reply:
The problem is that they are not necessarily going to be substituted and the length of the future name after a successful substitution is also unknown to the component.